cEP-0024.md

```diff
@@ -0,0 +1,129 @@
+# Implement GitMate automations and plugins for coala
+
+| Metadata |                                                       |
+| -------- | ----------------------------------------------------- |
+| cEP      | 24                                                    |
+| Version  | 1.0                                                   |
+| Title    | Implement Gitmate automations and plugins for coala   |
+| Authors  | Vamshi Krishna Bommerla <mailto:bommvams72@gmail.com> |
+| Status   | Proposed                                              |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the improvements to the coala workflow and the necessary
+automation to realise it, as a part of the
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5344316227059712).
+
+## Introduction
+
+[GitMate](https://gitmate.io/) is an automation tool for developers. While it
+works well, there are a number of missing features which would save time of
+coala community, for example automatically close PRs opened on master of forks.
+Many of the plugins are general plugins, which are required by every
+organization, but coala needs more plugins which support coala development
+workflow. This project adds plugins to auto-review or auto-reject PRs,
+assign issues.
+
+## New Features
+
+### Auto-assign issue when it is mentioned in any of PR commits
+
+When a PR is opened and if any of it's commit message mentions (Fixes, Closes)
+any unassigned issues and there is no CI failure, auto
+assign PR author to the mentioned unassigned issues. Comment on the
+PR (configurable) if someone else is working on the mentioned issue. If PR
+author is not a member of the organization, then he/she will be not be assigned
+to any of the issues.
+
+### Auto-reject PR opened on master of forks
+
+Normally, contributors should make changes in different branch from master
+branch of the forked repo, and submit a PR. But, sometimes coala receives PRs
+from master branch of the forked repo.
+
+coala needs mechanism by which GitMate automatically closes PRs opened on
+master branch of forked repo, marks certain label and comments on the PR
+informing the PR author to work on a different branch. Comment message, label
+will be configurable.
```

Comment:
what will the coala configuration be?

Add this everywhere in your cEP.

We can configure it after your project, but we should be able to let all maintainers read this cEP and agree with the initial implementation/configuration.